components/Buttons/src/MDCButton.m

```diff
@@ -305,7 +310,22 @@ - (void)layoutSubviews {
 }
 
 - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event {
-  return CGRectContainsPoint(UIEdgeInsetsInsetRect(self.bounds, _hitAreaInsets), point);
+  // If there are custom hitAreaInsets, use those
+  if (!UIEdgeInsetsEqualToEdgeInsets(self.hitAreaInsets, UIEdgeInsetsZero)) {
+    return CGRectContainsPoint(UIEdgeInsetsInsetRect(self.bounds, self.hitAreaInsets), point);
```

Comment:
Yes, it appears that UIEdgeInsetsInsetRect performs a literal manipulation as defined in the docs and does not standardize the rect first.